src/sentry/integrations/api/serializers/rest_framework/data_forwarder.py

```diff
@@ -0,0 +1,263 @@
+        s3_bucket = config.get("s3_bucket")
+        if s3_bucket is not None:
+            if not s3_bucket or s3_bucket.strip() == "":
+                errors.append("s3_bucket must be a non-empty string")
+            else:
+                s3_bucket_pattern = r"^[a-z0-9\-\.]+$"
+                if not re.match(s3_bucket_pattern, s3_bucket):
+                    errors.append("s3_bucket must be a valid S3 bucket name")
+
+    def _validate_segment_write_key(self, config: dict) -> None:
+        segment_write_key_pattern = r"^[a-zA-Z0-9_\-]+$"
+        write_key = config.get("write_key")
+        if not write_key or not re.match(segment_write_key_pattern, write_key):
+            raise ValidationError("write_key must be a valid Segment write key format")
+
+    def _validate_splunk_instance_url(self, config: dict, errors: list[str]) -> None:
+        splunk_url_pattern = r"^https?://[a-zA-Z0-9\-\.]+(?::\d+)?(?:/.*)?$"
+        instance_url = config.get("instance_URL")
+        if not instance_url or not re.match(splunk_url_pattern, instance_url):
+            errors.append("instance_URL must be a valid URL starting with http:// or https://")
+
+    def _validate_splunk_required_strings(self, config: dict, errors: list[str]) -> None:
+        index = config.get("index")
+        if not index or index.strip() == "":
+            errors.append("index must be a non-empty string")
+
+        source = config.get("source")
+        if not source or source.strip() == "":
+            errors.append("source must be a non-empty string")
+
+        token = config.get("token")
+        if not token or token.strip() == "":
+            errors.append("token must be a non-empty string")
+
+    def _validate_splunk_token_format(self, config: dict, errors: list[str]) -> None:
+        token = config.get("token")
+        if token and token.strip():  # Only validate if token exists and is not empty
+            splunk_token_pattern = r"^[a-zA-Z0-9\-]+$"
+            if not re.match(splunk_token_pattern, token):
+                errors.append("token must be a valid Splunk HEC token format")
+
+    def validate(self, attrs: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        organization_id = attrs.get("organization_id")
+        provider = attrs.get("provider")
+
+        if organization_id and provider:
+            existing = DataForwarder.objects.filter(
+                organization_id=organization_id, provider=provider
+            )
+
+            if self.instance:
+                existing = existing.exclude(id=self.instance.id)
+
+            if existing.exists():
+                raise ValidationError(
+                    f"A DataForwarder with provider '{provider}' already exists for this organization."
+                )
+
+        return attrs
+
+    def _validate_sqs_config(self, config) -> SQSConfig:
+        self._validate_all_fields_present(
+            config, ["queue_url", "region", "access_key", "secret_key"], "SQS"
```

Comment:
I think we should make the provider string an enum like in [IntegrationProviderSlug](https://github.com/getsentry/sentry/blob/crl/add-slack-send/src/sentry/integrations/types.py#L31), so that we know for sure the string will be normalized (i.e lowercase, uppercase etc.)